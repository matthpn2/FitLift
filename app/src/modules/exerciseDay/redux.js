import { createSelector } from 'reselect';

const initialState = {};

// selectors
const dbExercisesSelector = state => state.db.exercises;

export const exercisesMapper = createSelector(
  dbExercisesSelector,
  exercises => {
    if (Object.keys(exercises).length !== 0) {
      return Object.keys(exercises).reduce(
        (acc, x) => {
          const repsWeight = `${exercises[x].reps} ${exercises[x].weight}`;
          const sets = acc[exercises[x].type][repsWeight]
            ? (acc[exercises[x].type][repsWeight] += 1)
            : 1;
          return {
            ...acc,
            [exercises[x].type]: {
              ...acc[exercises[x].type],
              [repsWeight]: sets
            }
          };
        },
        {
          'Bicep Curls': {},
          'Single Front Raises': {},
          'Lateral Raises': {},
          'Tricep Extensions': {}
        }
      );
    }
    return null;
  }
);

export const exercisesSelector = createSelector(
  exercisesMapper,
  exercises =>
    exercises &&
    Object.keys(exercises).map(type => ({
      values: Object.keys(exercises[type]).map(x => ({
        reps: x.split(' ')[0],
        sets: exercises[type][x],
        type,
        weight: x.split(' ')[1]
      }))
    }))
);

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
