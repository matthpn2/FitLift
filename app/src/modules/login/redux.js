import { TypeKeys as apiTypeKeys } from '../../api/auth';

const initialState = {
  username: '',
  password: '',
  error: ''
};

export const TypeKeys = {
  UPDATE_INPUT: 'LOGIN:UPDATE_INPUT'
};

export const updateInput = (key, value) => ({
  type: TypeKeys.UPDATE_INPUT,
  key,
  value
});

export default (state = initialState, action) => {
  switch (action.type) {
    case TypeKeys.UPDATE_INPUT:
      return {
        ...state,
        [action.key]: action.value
      };
    case apiTypeKeys.LOGIN_FAIL:
      return {
        ...state,
        error: action.error
      };
    case apiTypeKeys.LOGIN_SUCCESS:
      return initialState;
    default:
      return state;
  }
};
