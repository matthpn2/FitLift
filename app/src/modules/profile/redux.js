import { createSelector } from 'reselect';
import reverse from 'lodash/reverse';
import firebase from '../../api/firebase';

const initialState = {};

export const TypeKeys = {
  DISPLAY_EXERCISE_DAY: 'DISPLAY_EXERCISE_DAY',
  LOGOUT: 'LOGOUT'
};

// selectors

const daysExercisedSelector = state => state.db.daysExercised;

export const daysExercisedArraySelector = createSelector(
  daysExercisedSelector,
  daysExercised => {
    if (Object.keys(daysExercised).length !== 0) {
      return reverse(Object.keys(daysExercised).map(x => x));
    }
    return null;
  }
);

// actions

export const displayExerciseDay = day => ({
  day,
  type: TypeKeys.DISPLAY_EXERCISE_DAY
});

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
