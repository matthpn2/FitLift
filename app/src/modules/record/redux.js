import { createSelector } from 'reselect';
import omit from 'lodash/omit';
import { TypeKeys as api } from '../../api/newExercises';

export const initialState = {
  isLoading: false,
  modifiedExercises: {}
};

export const TypeKeys = {
  LOGIN: 'login',
  RECEIVE_NEW_EXERCISES: api.RECEIVE_NEW_EXERCISES,
  REMOVE_NEW_EXERCISE: api.REMOVE_NEW_EXERCISE,
  REQUEST_NEW_EXERCISES: api.REQUEST_NEW_EXERCISES,
  UPDATE_NEW_EXERCISE: 'UPDATE_NEW_EXERCISE'
};

// actions

export const updateNewExercise = (id, key, value) => ({
  id,
  key,
  type: TypeKeys.UPDATE_NEW_EXERCISE,
  value
});

// selectors

const newExercisesSelector = state => state.db.newExercises;
const modifiedExercisesSelector = state => state.record.modifiedExercises;

export const exercisesToRecordSelector = createSelector(
  newExercisesSelector,
  modifiedExercisesSelector,
  (newExercises, modifiedExercises) => {
    if (Object.keys(newExercises).length !== 0) {
      return Object.keys(newExercises).map(x => ({
        ...newExercises[x],
        id: x,
        ...modifiedExercises[x]
      }));
    }
    return null;
  }
);

// reducer

export default (state = initialState, action) => {
  switch (action.type) {
    case TypeKeys.REQUEST_NEW_EXERCISES:
      return {
        ...state,
        isLoading: true
      };
    case TypeKeys.RECEIVE_NEW_EXERCISES:
      return {
        isLoading: false,
        modifiedExercises: {
          ...state.modifiedExercises,
          ...Object.keys(action.data).reduce(
            (acc, x) => ({
              ...acc,
              [x]: {
                reps: action.data[x].reps
              }
            }),
            {}
          )
        }
      };
    case TypeKeys.UPDATE_NEW_EXERCISE: {
      const { id, key, value } = action;
      const newValue = value ? value : undefined;
      return {
        ...state,
        modifiedExercises: {
          ...state.modifiedExercises,
          [id]: {
            ...state.modifiedExercises[id],
            [key]: newValue
          }
        }
      };
    }
    case TypeKeys.REMOVE_NEW_EXERCISE:
      return {
        ...state,
        modifiedExercises: omit(state.modifiedExercises, action.id)
      };
    case TypeKeys.LOGIN:
      return {
        ...state,
        isLoading: true
      };
    case TypeKeys.CONFIRMING_NEW_EXERCISE:
      return {
        ...state,
        modifiedExercises: {
          ...state.modifiedExercises,
          [action.id]: {
            ...state.modifiedExercises[action.id],
            isConfirming: true
          }
        }
      };
    default:
      return state;
  }
};
