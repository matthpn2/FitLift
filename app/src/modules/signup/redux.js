import { TypeKeys as apiTypeKeys } from '../../api/auth';

const initialState = {
  username: '',
  password: '',
  error: ''
};

export const TypeKeys = {
  UPDATE_INPUT: 'SIGNUP:UPDATE_INPUT'
};

export const updateInput = (key, value) => ({
  type: TypeKeys.UPDATE_INPUT,
  key,
  value
});

export default (state = initialState, action) => {
  switch (action.type) {
    case TypeKeys.UPDATE_INPUT:
      return {
        ...state,
        [action.key]: action.value
      };
    case apiTypeKeys.CREATE_USER_FAIL:
      return {
        ...state,
        error: action.error
      };
    case apiTypeKeys.CREATE_USER_SUCCESS:
      return initialState;
    default:
      return state;
  }
};
