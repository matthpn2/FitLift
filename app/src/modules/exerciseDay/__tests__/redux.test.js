import { exercisesSelector } from '../redux';

describe('selector tests', () => {
  it('should work', () => {
    const state = {
      db: {
        exercises: {
          '-L5wtK18VNAWE3sINej7': {
            reps: 5,
            timeStamp: 1519292600524,
            type: 'Bicep Curls',
            weight: '5'
          },
          '-L5wtKJ-6u_j8fCLt7dy': {
            reps: 5,
            timeStamp: 1519292601668,
            type: 'Bicep Curls',
            weight: '5'
          }
        }
      }
    };
    expect(exercisesSelector(state)).toMatchSnapshot();
  });
});
