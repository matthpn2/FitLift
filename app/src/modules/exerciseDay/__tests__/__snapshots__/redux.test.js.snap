// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selector tests should work 1`] = `
Array [
  Object {
    "values": Array [
      Object {
        "reps": "5",
        "sets": 2,
        "type": "Bicep Curls",
        "weight": "5",
      },
    ],
  },
  Object {
    "values": Array [],
  },
  Object {
    "values": Array [],
  },
  Object {
    "values": Array [],
  },
]
`;
